pkg/httpclient/client.go

```diff
@@ -43,29 +43,47 @@ func (c *Client) Do(r *http.Request) (resp *http.Response, err error) {
 	ctx := r.Context()
 
 	l := logging.WithComponent(ctx, c.component)
+	debug := l.Check(zap.DebugLevel, "debug") != nil
+
 	defer func() {
-		if err == nil {
+		if err != nil {
+			l.Error("error", zap.Error(err))
+			return
+		}
+		if debug {
 			l.Debug("outbound")
 			return
 		}
-		l.Error("error", zap.Error(err))
+		l.Info("outbound")
```

Comment:
Is this necessary? Or, was it left by mistake as `Info`?
It's not very useful as an info message and will only pollute the log

Reply:
This line will work when the log level is set to "info", so we can still have some logs.
Note that in "info" mode we do not dump the request/response but we only log basic info.

Reply:
I get that, but it's a "polluting" log line. It doesn't carry any context and it will just fill up our logs with "outbound" noise.

Reply:
It basically add only a checkpoint but that should be a development/debug behaviour not an info

Reply:
It will also add the fields that we defined "With".